test/optimisation/Optimisation.jl

```diff
@@ -635,6 +635,24 @@ using Turing
             maximum_a_posteriori(m; adtype=adbackend)
         end
     end
+
+    @testset "Collinear coeftable" begin
+        xs = [-1.0, 0.0, 1.0]
+        ys = [0.0, 0.0, 0.0]
+
+        @model function collinear(x, y)
+            a ~ Normal(0, 1)
+            b ~ Normal(0, 1)
+            return y ~ MvNormal(a .* x .+ b .* x, 1)
+        end
+
+        model = collinear(xs, ys)
+        mle_estimate = Turing.Optimisation.estimate_mode(model, MLE())
+        tab = coeftable(mle_estimate)
+        @assert isnan(tab.cols[2][1])
+        @assert tab.colnms[end] == "Error notes"
+        @assert occursin("singular", tab.cols[end][1])
+    end
```

Comment:
Could we also add a test where we check that for some harmless model where everything works out fine, `coeftable` returns the same result with and without `numerrors_warnonly`?

Also, a test case for negative variance would be great, to make sure tests cover all code paths.

Reply:
Yeah that's a good idea. Added! Agreed that a test with negative variance would be nice, but I'm not really sure how to construct a model which ends up with a negative variance estimate, and the original example where I got it is probably a bit heavy to add just for this. Since you offered to help finish off, maybe you could write this test? Thanks!

Reply:
Sure, I added a test for the negative variance case.